File: LibRetroRT/AV/GameGeometry.cs
namespace LibRetroRT.AV
{
    public struct GameGeometry
    {
        public uint BaseWidth;    /* Nominal video width of game. */
        public uint BaseHeight;   /* Nominal video height of game. */
        public uint MaxWidth;     /* Maximum possible width of game. */
        public uint MaxHeight;    /* Maximum possible height of game. */

        public float AspectRatio;  /* Nominal aspect ratio of game. If
                            * aspect_ratio is <= 0.0, an aspect ratio
                            * of base_width / base_height is assumed.
                            * A frontend could override this setting,
                            * if desired. */
    }
}

File: LibretroRT/GameGeometry.cpp
#include "pch.h"
#include "GameGeometry.h"

LibretroRT::GameGeometry::GameGeometry()
{

}

LibretroRT::GameGeometry::GameGeometry(GameGeometry ^ input)
{
	Initialize(input->BaseWidth, input->BaseHeight, input->MaxWidth, input->MaxHeight, input->AspectRatio);
}

LibretroRT::GameGeometry::GameGeometry(unsigned baseWidth, unsigned baseHeight, unsigned maxWidth, unsigned maxHeight, float aspectRatio)
{
	Initialize(baseWidth, baseHeight, maxWidth, maxHeight, aspectRatio);
}

void LibretroRT::GameGeometry::Initialize(unsigned baseWidth, unsigned baseHeight, unsigned maxWidth, unsigned maxHeight, float aspectRatio)
{
	BaseWidth = baseWidth;
	BaseHeight = baseHeight;
	MaxWidth = maxWidth;
	MaxHeight = maxHeight;
	AspectRatio = aspectRatio;
}
